docs/core/additional-tools/xml-serializer-generator.md

```diff
@@ -0,0 +1,131 @@
+> * How to edit your MyApp.csproj to add dependencies
+> * How to add a class and an XmlSerializer
+> * How to build and run the application 
+
+Like the [Xml Serializer Generator (sgen.exe)](https://docs.microsoft.com/en-us/dotnet/standard/serialization/xml-serializer-generator-tool-sgen-exe) for the .NET Framework, the Microsoft.XmlSerializer.Generator nuget package is the equivalent for .NET Core and .NET Standard projects. It creates an XML serialization assembly for types contained in an assembly to improve the startup performance of XML serialization when serializing or de-serializing objects of those types using <xref:System.Xml.Serialization.XmlSerializer>.
+
+## Prerequisites
+
+To complete this tutorial:
+
+* Install [.NET Core SDK 2.1.3 or later](https://www.microsoft.com/net/download)
+* Install your favorite code editor, if you haven't already.
+
+> [!TIP]
+> Need to install a code editor? Try [Visual Studio](https://aka.ms/vsdownload?utm_source=mscom&utm_campaign=msdocs)!
+  
+## Use Microsoft XML Serializer Generator in a .NET Core console application 
+
+The following instructions show you how to use XML Serializer Generator in a .NET Core console application.
+
+### Create a .NET Core console application
+
+Open a command prompt and create a folder named *MyApp*. Navigate to the folder you created and type the following command:
+
+```console
+dotnet new console
+```
+
+### Add a reference to the Microsoft.XmlSerializer.Generator package in the MyApp project
+
+Use the [`dotnet add package`](../tools//dotnet-add-package.md) command to add the reference in your project. 
+
+Type:
+ 
+ ```console
+ dotnet add package Microsoft.XmlSerializer.Generator -v 1.0.0
+ ```
+ 
+### Verify changes to MyApp.csproj after adding the package
+
+Open your code editor and let's get started! We're still working from the *MyApp* directory we built the app in.
+
+Open *MyApp.csproj* in your text editor.
+
+After running the [`dotnet add package`](../tools//dotnet-add-package.md) command, the following lines are added to your *MyApp.csproj* project file:
+
+ ```xml
+ <ItemGroup>
+    <PackageReference Include="Microsoft.XmlSerializer.Generator" Version="1.0.0" />
+ </ItemGroup>
+ ```
+ 
+### Add another ItemGroup section for .NET Core CLI Tool support
+ 
+ Add the following lines after the `ItemGroup` section that we inspected:
+ 
+ ```xml
+ <ItemGroup>
+    <DotNetCliToolReference Include="Microsoft.XmlSerializer.Generator" Version="1.0.0" />
+ </ItemGroup>
+ ```
+ 
+### Add a class in the application
+
+Open *Program.cs* in your text editor. Add the class named *MyClass* in *Program.cs*.
+
+```csharp
+public class MyClass
+{
+   public int Value;
+}
+```
+
+### Create an `XmlSerializer` for MyClass
+
+Add the following line inside *Main* to create an `XmlSerializer` for MyClass:
+
+```csharp
+var serializer = new System.Xml.Serialization.XmlSerializer(typeof(MyClass));
+```
+
+### Build and run the application
```

Comment:
should you show the whole file after all the steps are done so people can verify they did right?

Reply:
It's a dll.

Reply:
@JRAlexander I was talking about the source code above this step.